docs/reference-guides/data/data-core-edit-site.md

```diff
@@ -268,6 +268,18 @@ _Parameters_
 -   _options_ `[Object]`:
 -   _options.allowUndo_ `[boolean]`: Whether to allow the user to undo reverting the template. Default true.
 
+### setEditedPost
```

Comment:
Were the existing actions `setPage` / `setTemplate` not enough for the current use case here?

Reply:
`setTemplate` sets as edited entity the template and loads that specific view(in comparison to the page view with the resolved template for that page).

I tried this with both of the above and with `setPage` we don't have something to trigger the new template assigned and set it as the edited entity(as the postId and postType remain the same). I might have missed something though 🤷 .

Reply:
Ok. I actually think the problem comes from how we're structuring our reducer. 

In other words, setting the template of a post, should be enough to tell the site editor to actually reload with the new template. The problem is that we have this "state" in two places so we need to synchronize.

In an ideal scenario, I think the site editor reducer (state) shouldn't contain any reference to the "template". It should only contain a reference the main edited entity (the one in the url). So for instance when a page or post is opened, it should only contain a reference to the postId and postType. And the template should be retrieved using regular "core-data" selectors when needed.

That said, this is definitely out of scope of the current PR and something I've noticed before. I think we would benefit (simplification) from such a refactoring at some point.

---

That said, it's a bit weird that `setPage` doesn't take the edited entity into consideration.

Reply:
I had the same question. Besides the extra requests that `setPage` does, the actions look the same.

P.S. Should we stick with the current naming convention and use `setEditedPage` as the name?

Reply:
>That said, it's a bit weird that setPage doesn't take the edited entity into consideration.

After trying things until I get the flow it seems the issue was with `resolveSelect` inside `setPage` action.
For now I replaced it with `apiFetch` call. Any better suggestions? 🤔